### DISCUSSION TITLE
[Probably issue] Preheat Material icons are not proper displayed

### DISCUSSION THREAD

**Discussion by Giadej** [Jyers/Marlin#622] [Q&A] [answered]
Hi all,
does it happen to any of you that the material preheating icons are not displayed correctly?
I am attaching a photo of what I mean

![tempmaterial](https://user-images.githubusercontent.com/80910217/116210269-fd0b7a00-a742-11eb-83f4-11bf896c2666.jpeg)

is this only my problem and/or is there a solution?

**Comment by gothcha** [comment]
This is how it should look I believe. It is certainly how my display is.

**Comment by Ali229** [comment]
Look pretty normal to me, what's not correct?

**Reply by Giadej** [reply]
@gothcha @Ali229 
This is as shown in the original firmware:

![ori](https://user-images.githubusercontent.com/80910217/116253866-2642ff00-a771-11eb-82ef-d05ef21445c5.jpeg)

**Reply by Giadej** [reply]
the proper icon are already incuded in 9.ico iconset

**032-ICON_ABSPreheat** ![032-ICON_ABSPreheat](https://user-images.githubusercontent.com/80910217/116255009-21cb1600-a772-11eb-96bc-12d0b17af021.jpg) 

**031-ICON_PLAPreheat** ![031-ICON_PLAPreheat](https://user-images.githubusercontent.com/80910217/116255012-21cb1600-a772-11eb-92e5-6c0c59a57337.jpg)

**Reply by tome9111991** [reply]
![tempmaterial](https://user-images.githubusercontent.com/80910217/116210269-fd0b7a00-a742-11eb-83f4-11bf896c2666.jpeg)


> It default in jyers.
> 
> The problem user can set own preheat setting.
> Atm you can only configure a icon for Preheat label.
> 
> in this case.
> PLA ist Preheat label 2.
> You can set a PLA icon for Preheat label 2.
> But is your label 2 not PLA it will show PLA icon.
> 
> Sorry for bad english 😅

**Comment by smooreace** [comment]
So they are displaying fine, their just different?

**Reply by gothcha** [reply]
Apologies there are a few 9.ICO files in use. I am using the @AngleV set with the colour main menu Icons, but this does not have the PLA,ABS etc in it.

<img width="477" alt="Screenshot 2021-04-27 at 15 51 55" src="https://user-images.githubusercontent.com/80574048/116262528-86857100-a770-11eb-865c-37aa8ad358a0.png">

@Jyers are you intending to combine the best of the different Icons sets?

**Reply by Giadej** [reply]
me too

**Reply by Jyers** [reply]
@gothcha Once I finally get the display firmware folder organized I'll set up different folders for the different icon packs for people to choose from.

**Reply by gothcha** [reply]
@Jyers good plan, and I agree with your logic re the PLA icons, I am more than happy with the default icon, the text says it all!

**Reply by Giadej** [reply]
> 
> 
> @gothcha
> Just to point out to you the attribution of your work on the icons by @Giadej and its distribution under @Giadej for mriscoc, famous developer who does not share anything but takes everything.

Relax yourself young boy, you're embarrassing with your hate shooting on everyone!!!
I gave everyone the right credits!

**AngelV** for "Main Panel" icons
**Lighty1989** for "XYZ Move Axis" icons and "Bed Points" icons
**Gotcha's** for other icons
**Giadej** for custom Boot-Screen

https://github.com/mriscoc/Marlin_Ender3v2/blob/Ender3v2-Released/display%20assets/Giadej%20compilation/readme.md

**Reply by Giadej** [reply]
> 
> 
> @Giadej
> I am only targeting Mriscoc (and I am not the only one)[...]
> Hoping that Google translate does not alter what I write ....

Sorry but from the way you posed with malice it seemed that your attack was referring to me.
These developer wars aren't good for end users like me. We use your work and thank you for the work done.
As regards the licensing problem, there are checks in progress and one has to wait to understand any discrepancies.
As for donations, I can only tell you that you are wrong: all experimental releases on mriscoc's Telgram channel are FREE.
The end user is free to support the developer with donations.

**Comment by Jyers** [comment, marked as answer]
Because all of the preheat options are customizable when compiling (even the labels) I chose not to use those icons. Having a PLA icon wouldn't make much sense if you switched it to say TPU...

**Reply by Giadej** [reply]
We have 3 options (PLA, ABS, PETG) in preheat submenu. Two icons already exist in iconset, there would be a need only PETG one.

**Reply by tome9111991** [reply]
> Because all of the preheat options are customizable when compiling (even the labels) I chose not to use those icons. Having a PLA icon wouldn't make much sense if you switched it to say TPU...



Maybe an idea how to use customizable icons

Add in configuration
#define PREHEAT_2_Icon  ICON_PLAPreheat

In Dwin.cpp

Draw_Menu_Item(row, PREHEAT_2_Icon, (char*)PREHEAT_2_LABEL);


maybe someone will get it to work that way

**Reply by Jyers** [reply]
I don't really think thats worth the extra define. Adding things to the config files is not ideal and makes it harder for people to adapt their own configs. I think the temp icons look fine. Thanks for the thoughts though

**Reply by tome9111991** [reply]
> I don't really think thats worth the extra define. Adding things to the config files is not ideal and makes it harder for people to adapt their own configs. I think the temp icons look fine. Thanks for the thoughts though

@Jyers 
customizable icons per Config
is just a test but looks like it works
The problem is everyone has to update their config otherwise they will get an error

![20210511_125430](https://user-images.githubusercontent.com/57866234/117805787-aed59b00-b259-11eb-9716-039e13d955ac.jpg)

![config](https://user-images.githubusercontent.com/57866234/117805858-c3b22e80-b259-11eb-9e3e-1877c46568d4.png)
![dwin](https://user-images.githubusercontent.com/57866234/117805862-c44ac500-b259-11eb-9bfe-237a879119a1.png)




Edit:
For exemple this should be fix some erros.

 #if ENABLED(PREHEAT_2_ICON)
                 Draw_Menu_Item(row, PREHEAT_2_ICON, (char*)PREHEAT_2_LABEL, NULL, true);
 #else
                   Draw_Menu_Item(row, ICON_Temperature, (char*)PREHEAT_2_LABEL, NULL, true);
 #endif
